build-tools/src/main/kotlin/org/jetbrains/kotlin/KonanTestExecutable.kt

```diff
@@ -0,0 +1,10 @@
+package org.jetbrains.kotlin
+
+import org.gradle.api.Action
+import org.gradle.api.Task
+
+interface KonanTestExecutable : Task {
+    val executable: String
+    var doBeforeRun: Action<in Task>?
+    var doBeforeBuild: Action<in Task>?
```

Comment:
This is somewhat unreadable.
Do I understand correctly that its usages rely on some kind of Groovy/Gradle convention, which makes `doBeforeBuild { ... }` mean assigning this lambda to `doBeforeBuild`? If so, this definitely requires some comments added.

Reply:
Yep, I'm going to add a doc to this. As soon as most tests are executed and built in different tasks, there should be a some kind of convention to add actions to be done before build or before the run but after the build. The order is the following:  `doBeforeBuild {} -> build test executable -> doBeforeRun { } -> run test executable`